server/src-lib/Hasura/GraphQL/Validate/Types.hs

```diff
@@ -145,9 +149,10 @@ type ObjFieldMap = Map.HashMap G.Name ObjFldInfo
 
 data ObjTyInfo
   = ObjTyInfo
-  { _otiDesc   :: !(Maybe G.Description)
-  , _otiName   :: !G.NamedType
-  , _otiFields :: !ObjFieldMap
+  { _otiDesc       :: !(Maybe G.Description)
+  , _otiName       :: !G.NamedType
+  , _otiImplIFaces :: [G.NamedType]
```

Comment:
Maybe we can use a hashset here?

Reply:
Sure. Changed from Array of implementing interfaces to HashSet.